pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -65,208 +65,376 @@ var reInvalidUIDCharacter = regexp.MustCompile(`[^a-z0-9\-]`)
-				klog.Errorf("Unable to retrieve most recent metrics: %v", err)
-				return []record.Record{{Name: "config/metrics", Item: RawByte(fmt.Sprintf("# error: %v\n", err))}}, nil
-			}
-			return []record.Record{
-				{Name: "config/metrics", Item: RawByte(data)},
-			}, nil
-		},
-		func() ([]record.Record, []error) {
-			config, err := i.client.ClusterOperators().List(metav1.ListOptions{})
-			if errors.IsNotFound(err) {
-				return nil, nil
-			}
-			if err != nil {
-				return nil, []error{err}
-			}
-			records := make([]record.Record, 0, len(config.Items))
-			for i := range config.Items {
-				records = append(records, record.Record{Name: fmt.Sprintf("config/clusteroperator/%s", config.Items[i].Name), Item: ClusterOperatorAnonymizer{&config.Items[i]}})
-			}
-			namespaceEventsCollected := sets.NewString()
+		GatherMostRecentMetrics(i),
+		GatherClusterOperators(i),
+		GatherUnhealthyNodes(i),
+		GatherConfigMaps(i),
+		GatherClusterVersion(i),
+		GatherClusterID(i),
+		GatherClusterInfrastructure(i),
+		GatherClusterNetwork(i),
+		GatherClusterAuthentication(i),
+		GatherClusterFeatureGates(i),
+		GatherClusterOAuth(i),
+		GatherClusterIngress(i),
+		GatherClusterProxy(i),
+		GatherCertificateSigningRequests(i),
+	)
+}
 
-			now := time.Now()
-			for _, item := range config.Items {
-				if isHealthyOperator(&item) {
+// GatherMostRecentMetrics gathers cluster Federated Monitoring metrics
+//
+// The GET REST query to URL /federate
+// Gathered metrics:
+//   ALERTS
+//   etcd_object_counts
+//   cluster_installer
+//
+// Location of operators in archive: config/metrics/
+// see: docs/insights-archive-sample/config/metrics
+func GatherMostRecentMetrics(i *Gatherer) func() ([]record.Record, []error) {
+	return func() ([]record.Record, []error) {
+		if i.metricsClient == nil {
+			return nil, nil
+		}
+		data, err := i.metricsClient.Get().AbsPath("federate").
+			Param("match[]", "ALERTS").
+			Param("match[]", "etcd_object_counts").
+			Param("match[]", "cluster_installer").
+			DoRaw()
+		if err != nil {
+			// write metrics errors to the file format as a comment
+			klog.Errorf("Unable to retrieve most recent metrics: %v", err)
+			return []record.Record{{Name: "config/metrics", Item: RawByte(fmt.Sprintf("# error: %v\n", err))}}, nil
+		}
+		records := []record.Record{
+			{Name: "config/metrics", Item: RawByte(data)},
+		}
+
+		return records, nil
+	}
+}
+
+// GatherClusterOperators collects all ClusterOperators
+// In unhealthy operators it finds unhealthy Pods
+//
+// The Kubernetes api https://github.com/openshift/client-go/blob/master/config/clientset/versioned/typed/config/v1/clusteroperator.go#L62
+// Response see https://docs.openshift.com/container-platform/4.3/rest_api/index.html#clusteroperatorlist-v1config-openshift-io
+//
+// Location of operators in archive: config/clusteroperator/
+// see: docs/insights-archive-sample/config/clusteroperator
+// Location of pods in archive: config/pod/
```

Comment:
No example for pods?

Reply:
I would like to add more samples later. I just have few samples ready and it takes some time to get each prepared with the test.